src/teleop_twist_joy.cpp

```diff
@@ -46,188 +46,156 @@ namespace teleop_twist_joy
 struct TeleopTwistJoy::Impl
 {
   void joyCallback(const sensor_msgs::msg::Joy::SharedPtr joy);
+  void sendCmdVelMsg(const sensor_msgs::msg::Joy::SharedPtr, const std::string& which_map);
 
-  rclcpp::Node::SharedPtr node;
-
-  rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr cmd_vel_pub;
   rclcpp::Subscription<sensor_msgs::msg::Joy>::SharedPtr joy_sub;
+  rclcpp::Publisher<geometry_msgs::msg::Twist>::SharedPtr cmd_vel_pub;
 
   int enable_button;
   int enable_turbo_button;
 
   std::map<std::string, int> axis_linear_map;
-  std::map<std::string, double> scale_linear_map;
-  std::map<std::string, double> scale_linear_turbo_map;
+  std::map<std::string, std::map<std::string, double>> scale_linear_map;
 
   std::map<std::string, int> axis_angular_map;
-  std::map<std::string, double> scale_angular_map;
-  std::map<std::string, double> scale_angular_turbo_map;
+  std::map<std::string, std::map<std::string, double>> scale_angular_map;
 
   bool sent_disable_msg;
 };
 
 /**
  * Constructs TeleopTwistJoy.
- * \param nh NodeHandle to use for setting up the publisher and subscriber.
- * \param nh_param NodeHandle to use for searching for configuration parameters.
  */
-TeleopTwistJoy::TeleopTwistJoy(rclcpp::Node::SharedPtr & node)
+TeleopTwistJoy::TeleopTwistJoy() : Node("teleop_twist_joy_node")
 {
-  pimpl_ = new Impl();
-
-  pimpl_->node = node;
+  pimpl_ = new Impl;
```

Comment:
Consider using smart pointer instead.

Reply:
Yeah, I went back and forth on that.  I ended up leaving it as a raw pointer to be more compatible with upstream https://github.com/ros-teleop/teleop_twist_joy, which has to support all the way back to Indigo.  So I think I'll leave this one as-is as well.